compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,270 @@
+    .default_value(false)
+    .help("Show version information and exit")
+    .exit_with(print_version);
+
+  // TODO Add new options!
+
+  arser.add_argument("--input").nargs(1).type(arser::DataType::STR).help("Input circle model");
+  arser.add_argument("--output").nargs(1).type(arser::DataType::STR).help("Output circle model");
+
+  // select option
+  arser.add_argument("--by_id")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation id to select nodes.");
+  arser.add_argument("--by_name")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation name to select nodes.");
+  arser.add_argument("--select")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Selecte opeartors from the input circle");
+  arser.add_argument("--deselect")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Exclude operators from the input circle");
+
+  try
+  {
+    arser.parse(argc, argv);
+  }
+  catch (const std::runtime_error &err)
+  {
+    std::cout << err.what() << std::endl;
+    std::cout << arser;
+    return EXIT_FAILURE;
+  }
+
+  std::string input_path = arser.get<std::string>("--input");
+  std::string output_path = arser.get<std::string>("--output");
+
+  std::string operator_input;
+
+  std::vector<int> by_id;
+  std::vector<std::string> by_name;
+
+  std::string op;
+  std::vector<int> oplist;
+  int select_mode = -1;
+
+  if (arser["--by_id"])
+  {
+    operator_input = arser.get<std::string>("--by_id");
+    split_id_input(operator_input, by_id);
+  }
+  if (arser["--by_name"])
+  {
+    operator_input = arser.get<std::string>("--by_name");
+    split_name_input(operator_input, by_name);
+  }
+  if (arser["--select"])
+  {
+  }
+  if (arser["--deselect"])
+  {
+  }
+
+  // option parsing test code.
+  for (int x : by_id)
+    std::cout << "by_id: " << x << std::endl;
+
+  for (std::string line : by_name)
+    std::cout << "by_name: " << line << std::endl;
+
+  // Load model from the file
+  foder::FileLoader file_loader{input_path};
+  std::vector<char> model_data = file_loader.load();
+
+  // Verify flatbuffers
+  flatbuffers::Verifier verifier{reinterpret_cast<uint8_t *>(model_data.data()), model_data.size()};
+  if (!circle::VerifyModelBuffer(verifier))
+  {
+    std::cerr << "ERROR: Invalid input file '" << input_path << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  const circle::Model *circle_model = circle::GetModel(model_data.data());
+  if (circle_model == nullptr)
+  {
+    std::cerr << "ERROR: Failed to load circle '" << input_path << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Import from input Circle file
+  luci::Importer importer;
+  auto module = importer.importModule(circle_model);
+
+  // Select and Import from user input.
+  opselector::OpSelector *selector = new opselector::OpSelector(circle_model);
+  std::map<uint32_t, std::string> _source_table = module.get()->source_table();
+  std::map<uint32_t, std::string> id_name_selected_nodes;
```

Comment:
I think there is a misunderstanding here. Please see the comment in https://github.com/Samsung/ONE/issues/7662#issuecomment-917644516.
> All you should do is just look at the imported module(graphs) and create a new graph based on that information. Actually, imported module has all the information you want. For example, `name` is already included in `CircleNode` class, which can be accessed by `name` method. `index` can be accessed by `get_node_id` method(@llFreetimell This is the only way to get the id, isn't it?).  

And if you want to know about this more detail, see https://github.com/Samsung/ONE/issues/6080.

In short, `source_table` does not represent the model that has it, but has the data of the model's original model. So, again, you should use above `module`.